### PR TITLE
Documentation maintenance: tooling update and metadata fixes.

### PR DESCRIPTION
**Pull request by ADubhlaoich**
### Proposed changes

This pull request addresses a number of problems with the documentation:

- The documentation tooling (Makefile and tool-specific configuration) is updated to the current standard
- The guidance on Hugo and Markdown formatting has been updated in the documentation subfolder README
- Every individual Markdown file related to the documentation has had its unnecessary metadata removed
- Where relevant, the opening paragraph for each document has been changed or removed by moving the frontmatter description or removing duplicate text

There will more broad documentation improvements in the future, but this pull request cleans up the most egregious issues which will make tooling and content improvements much easier, as all issues addressed are blockers for these efforts.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -37,68 +39,50 @@ ifneq (, $(shell $(NETLIFY) "true"))
...
-
-# Can be used to deploy to netlify from your local
-# development environment.
-# Requires a netlify login.
-netlify: clean
-	netlify deploy --build -d public --alias $(shell git branch --show-current)-branch
-
 deploy-preview: hugo-mod
-	hugo --gc -b ${NETLIFY_DEPLOY_URL}/nginx-ingress-controller
+	hugo --gc -b ${NETLIFY_DEPLOY_URL}/nginx-gateway-fabric/
```
Comment (j1m-ryan) [inline docs/Makefile:88]: this should be `nginx-ingress-controller` I imagine?

Comment (ADubhlaoich) [inline docs/Makefile:88]:
```suggestion
	hugo --gc -b ${NETLIFY_DEPLOY_URL}/nginx-ingress-controller/
```

File: docs/content/configuration/ingress-resources/basic-configuration.md [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
-weight: 1600
-doctypes: [""]
 toc: true
-docs: "DOCS-593"
+weight: 1600
 ---
 
-
-The example below shows a basic Ingress resource definition. It load balances requests for two services -- coffee and tea -- comprising a hypothetical *cafe* app hosted at `cafe.example.com`:
+This document showss a basic Ingress resource definition which load balances requests for two services as part of a single application.
```
Comment (j1m-ryan) [inline docs/content/configuration/ingress-resources/basic-configuration.md:11]:
```suggestion
This document shows a basic Ingress resource definition which load balances requests for two services as part of a single application.
```

File: docs/content/installation/nic-images/pulling-ingress-controller-image.md [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
-description: "Learn how to pull an F5 NGINX Plus Ingress Controller image—including those with NGINX App Protect WAF and NGINX App Protect DoS—from the official F5 Docker registry and upload it to your private registry. This guide covers the prerequisites, image tagging, and troubleshooting steps."
-weight: 100
-doctypes: ["install"]
 toc: true
-docs: "DOCS-605"
+weight: 100
 ---
 
-{{<custom-styles>}}
+Learn how to pull an F5 NGINX Plus Ingress Controller image those with NGINX App Protect WAF and NGINX App Protect DoS from the official F5 Docker registry and upload it to your private registry.
```
Comment (j1m-ryan) [inline docs/content/installation/nic-images/pulling-ingress-controller-image.md:10]:
```suggestion
Learn how to pull an F5 NGINX Plus Ingress Controller image, including those with NGINX App Protect WAF and NGINX App Protect DoS, from the official F5 Docker registry and upload it to your private registry. 
```

**Comment by ADubhlaoich** [comment]
Closed in favour of #5328 to bypass a permissions problem.

**Pull request closed without merging**